common/lib/capa/capa/capa_problem.py

```diff
@@ -846,7 +846,7 @@ def _preprocess_problem(self, tree):  # private
             answer_id = 1
             input_tags = inputtypes.registry.registered_tags()
             inputfields = tree.xpath(
-                "|".join(['//' + response.tag + '[@id=$id]//' + x for x in input_tags + solution_tags]),
```

Comment:
@cahrens `solution_tags` here is incorrect. `solution_tags` is being treated as inputtype in the above statement which is wrong. Due to this, problem rendering was failing if we try to put `<solution>` inside a `responsetype`. Now after the above change we are able to put `<solution>` inside a `responsetype`